### DISCUSSION TITLE
TX to Suntouch always aborts after 4 attempts

### DISCUSSION THREAD

**Discussion by virtualuk** [tagyoureit/nodejs-poolController#777] [App and O/S Config Questions] [answered]
I'm trying to run njspc with dashboard (and ultimately with the HA HACS custom component).  The njspc software is running on an R-PI3 with the latest R-OS for ARM7, with npm and node versions shown as 9.6.7 and v18.16.0 respectively. My pool equipment is a Pentair Suntouch (that I'm hooking the RS485 into) and a Pentair Intelliflow VS-3050 pump.  RS485 is set to 

I've tried a couple of different RS485 USB sticks including the recommended ones but I get the same behavior.  My instance of njspc can detect a controller and it can show stats like temperature,  can show if the pump is active, the RPMs and wattage, but if I try to update any settings connected to the pump I get a pop-up and corresponding console message with an OutboundMessageError that says "Message aborted after 4 attempt(s)". Screenshot attached of when I tried to set the pool capacity. 

One thing I did notice, and not sure if this is part of it, but under the controller section on the dashboard it doesn't show any details (, although it correctly shows in the title bar SunTouch as the model).  I've included some images from the RPi setup. 

Thanks in advance for any help!

![20230605_153233](https://github.com/tagyoureit/nodejs-poolController/assets/8462061/9383f56f-4acf-4fca-80b2-a58c0e558943)
![20230605_153316](https://github.com/tagyoureit/nodejs-poolController/assets/8462061/94325750-2069-4945-ae40-98600497a3c4)

**Comment by rstrouse** [comment]
Ok we need a replay that includes the reload configuration.  It appears we are requesting information that SunTouch does not return.  For instance the 168 message sets the heating priority for the spa.  It looks like that function does not exist in SunTouch.

**Comment by virtualuk** [comment]
Here's the replay. After starting the capture with configuration reload it turned the model in Nixie Single Body, is that normal?
# #
[replay (1).zip](https://github.com/tagyoureit/nodejs-poolController/files/11658504/replay.1.zip)

**Reply by rstrouse** [reply]
No that is not normal.  It should have picked up the information from the status messages coming from the RS485 port.  I'm looking into it.

**Reply by rstrouse** [reply]
Can I get you to do this again while the controller thinks it is a Nixie controller?

**Reply by virtualuk** [reply]
Here's the new replay.  If it helps, I can blow away the install and start from scratch to give you a clean view too if needed.
[replay (2).zip](https://github.com/tagyoureit/nodejs-poolController/files/11658812/replay.2.zip)

**Reply by rstrouse** [reply]
No I need to know why it isn't picking up the SunTouch data from the RS485.

**Reply by virtualuk** [reply]
What's strange was the Nixie Single Body only showed up as soon as I enabled the first replay. Prior to that the model information was blank, but it had SunTouch in the title bar.  Let me know if there's anything else I can get you to help.

**Reply by rstrouse** [reply]
Is this capture replay with the Configuration Reload checked?

**Reply by virtualuk** [reply]
Come to think of it, that did seem a bit fast for the reload.  To make sure, I did it again, but it hung on the initialization stage so I bounced the server and dashboard and did another replay making sure the configuration reload was checked,  I also captured the console messages from the server during startup and while I was running the capture.
[replay.zip](https://github.com/tagyoureit/nodejs-
[njspc-startup202306051852.txt](https://github.com/tagyoureit/nodejs-poolController/files/11659175/njspc-startup202306051852.txt)
poolController/files/11659171/replay.zip)

**Reply by virtualuk** [reply]
Hmmm here's the replay.zip
[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/11659213/replay.zip)

**Reply by virtualuk** [reply]
and here's the console log from startup 
[njspc-startup202306051852.txt](https://github.com/tagyoureit/nodejs-poolController/files/11659217/njspc-startup202306051852.txt)

**Reply by rstrouse** [reply]
Pull njsPC.  I modified the controller reset to allow detection for the SunTouch series boards.  Am I seeing this correctly do you have a shared system pool/spa?

After you update njsPC please perform the replay with a configuration reload again.  I need to see what is responded to by the controller.

**Reply by virtualuk** [reply]
Thanks for the update, I pulled the latest and captured a replay with the configuration reload.  My setup is just the pool, originally the pool guy had the pool sweep setup on spa using actuators to divert flow to the sweep and ramping the RPMs up.  That's probably why both pool and spa are showing up, I was hoping with njspc I could do something a little more sophisticated perhaps :) 

The replay with with the configuration reload and attempting to set the pool body size.  Thanks for the help!
[replay (1).zip](https://github.com/tagyoureit/nodejs-poolController/files/11674261/replay.1.zip)

**Reply by rstrouse** [reply]
That looks like a much better output.  At least it waited long enough to get a response from SunTouch before assuming it did not exist.  I see several requests to the panel that are being made for functionality that probably does not exist in SunTouch but exists in the Touch protocol overall so I have a few questions.

1. Does SunTouch have an option for Manual heat related to spa heaters?
2. Does SunTouch support heatpumps?
3. Can you set a circuit function in SunTouch for instance (Master Cleaner, IntelliBrite ... etc)
4. Can valves be assigned to a circuit/feature?
5. Does the panel support dual speed pumps?
6. Is there an option to enable/disable IntelliChlor?

**Reply by virtualuk** [reply]
Hopefully this helps:

1) Not sure if it's specific to spa heaters.  In the Heat menu from the Main Menu there is
 Main Menu
  -> Heat
    ->Pool
      -> Temp (set temperature)
      ->Heat (on/off)
    -> Spa
      -> Temp (set temperature)
      -> Heat (on/off)
    -> Manual Heat
      -> Manual Heat (on/off)

BTW - I don't have any heater on the pool other than the solar heating on the roof.

2) I don't see anything in any of the menus for heatpumps

3) Available circuit functions are
Main Menu
  -> Circ. functions
    -> Pool Circuit
      -> Freeze (on/off)
    -> Spa Circuit
      -> Freeze (on/off)
    -> Aux1 Circuit
      -> Generic, Solar, Heater, Floor clnr, Spillway, Cleaner
      -> Freeze (on/off)
    -> Aux2 Circuit
      -> Generic, Solar, Heater, Floor clnr, Spillway, Cleaner
      -> Freeze (on/off)
    -> Aux3 Circuit
      -> Generic, Solar, Heater, Floor clnr, Spillway, Cleaner
      -> Freeze (on/off)

4) Yes, it has
Main Menu
  -> Valve Circuit
    -> Solar, Feature 4, Feature 3, Feature 2, Feature 1, Aux3, Aux2, Aux1, Pool, Spa, None, Freeze, Heater, 

5)  Yes, from the menu it looks like it supports single, dual, Intelliflow VF and Intellflow VS

6) Yes

**Reply by rstrouse** [reply]
How old is this panel?  All the queries sent to it to get the configuration are summarily ignored.  Even the time on the panel says it it 2009 and it doesn't report the firmware version.

**Reply by virtualuk** [reply]
Managed to find a sticker on the circuit board with a manufacturer date of 1/25/2011 and the firmware version is 2.060

**Comment by virtualuk** [comment]
![20230607_190702](https://github.com/tagyoureit/nodejs-poolController/assets/8462061/e859cece-2899-4301-9708-b71bf7b8d026)

**Reply by rstrouse** [reply]
Ok so I think I know what is going on.  The earliest firmware that I see that emits the config is 2.070.  The current version is 2.090.  I think there are a couple of choices with this.  Either get the firmware flashed to a later version or use the power supply and internal relays of the  SunTouch while replacing the controller with the pi and a Nixie controller.

https://www.pentair.com/en-us/education-support/residential/product-support/pentair-pool-and-spa-software-downloads/intellitouch-easytouch-suntouch-ilink-firmware.html

**Reply by virtualuk** [reply]
Ahhh ok, I didn't know updating the firmware was a thing :)  I'll have to see about how to take care of that and report back, thanks!

**Reply by virtualuk** [reply]
Hmm having the firmware is one thing, knowing how to get it onto the controller is another, not much info out there it looks like.  Haven't looked at the Nixie approach either but sounds like a Pentair free (ie not selling a kidney to fix it) solution :)

**Reply by rstrouse** [reply]
It is definitely waaaaaay more powerful than SunTouch and will give you better control over the equipment.  To update the SunTouch firmware you either:
1. Have to buy a new board with the new firmware on it
2. Send the board to somebody who can flash it for you
3. Buy a programmer (these are stupid expensive)
4. Build a programmer from cobbled parts (A couple of folks on TFP did this awhile back)

I still believe your best option is Nixie.

**Reply by virtualuk** [reply]
I'm thinking so too after skimming through the nixie docs. I'm not using any relays and really just need to connect RS485 to the pump with the pi that I'm already running njspc and dashboard on.  Is it really as simple as disconnecting the SunTouch from the Intelliflow VS and connecting the RS485 direct to the pump? That seems too good to be true :)

**Reply by rstrouse** [reply, marked as answer]
That is all you need

**Reply by virtualuk** [reply]
Dang... disconnected the comms between the pool and the SunTouch, and hey presto, I have a full controllable pool setup.  I even have it hooked into Home Assistant, super happy! :)  Thank you for all the help!

**Reply by BlackAlgae** [reply]
@virtualuk hi there, did you flash the firmware and then disconnected the comms? or just disconnected the comms in your suntouch and it worked for you. Thank you
